<a href="https://colab.research.google.com/github/nayoung718/essa_python_1/blob/main/4_12_%EC%84%B8%EC%85%98_%ED%85%8D%EC%8A%A4%ED%8A%B8%EB%B6%84%EC%84%9D_%EC%97%B0%EC%8A%B5%EB%AC%B8%EC%A0%9C.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1. 아래의 데이터를 이용하여 공부한 내용을 바탕으로 문제 2개를 만들고 답하세요.

In [1]:
import nltk
nltk.download("book", quiet=True)
from nltk.book import *

*** Introductory Examples for the NLTK Book ***
Loading text1, ..., text9 and sent1, ..., sent9
Type the name of the text or sentence to view it.
Type: 'texts()' or 'sents()' to list the materials.
text1: Moby Dick by Herman Melville 1851
text2: Sense and Sensibility by Jane Austen 1811
text3: The Book of Genesis
text4: Inaugural Address Corpus
text5: Chat Corpus
text6: Monty Python and the Holy Grail
text7: Wall Street Journal
text8: Personals Corpus
text9: The Man Who Was Thursday by G . K . Chesterton 1908


In [2]:
# 파일 목록 확인
nltk.corpus.gutenberg.fileids()

['austen-emma.txt',
 'austen-persuasion.txt',
 'austen-sense.txt',
 'bible-kjv.txt',
 'blake-poems.txt',
 'bryant-stories.txt',
 'burgess-busterbrown.txt',
 'carroll-alice.txt',
 'chesterton-ball.txt',
 'chesterton-brown.txt',
 'chesterton-thursday.txt',
 'edgeworth-parents.txt',
 'melville-moby_dick.txt',
 'milton-paradise.txt',
 'shakespeare-caesar.txt',
 'shakespeare-hamlet.txt',
 'shakespeare-macbeth.txt',
 'whitman-leaves.txt']

In [3]:
hamlet = nltk.corpus.gutenberg.raw('shakespeare-hamlet.txt')
print(hamlet[:10000])

[The Tragedie of Hamlet by William Shakespeare 1599]


Actus Primus. Scoena Prima.

Enter Barnardo and Francisco two Centinels.

  Barnardo. Who's there?
  Fran. Nay answer me: Stand & vnfold
your selfe

   Bar. Long liue the King

   Fran. Barnardo?
  Bar. He

   Fran. You come most carefully vpon your houre

   Bar. 'Tis now strook twelue, get thee to bed Francisco

   Fran. For this releefe much thankes: 'Tis bitter cold,
And I am sicke at heart

   Barn. Haue you had quiet Guard?
  Fran. Not a Mouse stirring

   Barn. Well, goodnight. If you do meet Horatio and
Marcellus, the Riuals of my Watch, bid them make hast.
Enter Horatio and Marcellus.

  Fran. I thinke I heare them. Stand: who's there?
  Hor. Friends to this ground

   Mar. And Leige-men to the Dane

   Fran. Giue you good night

   Mar. O farwel honest Soldier, who hath relieu'd you?
  Fra. Barnardo ha's my place: giue you goodnight.

Exit Fran.

  Mar. Holla Barnardo

   Bar. Say, what is Horatio there?
  Hor. A peece of

- VADER를 이용한 감성 분석

VADER : 소셜 미디어의 감성 분석 용도로 만들어진 룰 기반의 Lexicon

### (1) 문제 1 : 주어진 텍스트 데이터를 사용하여 VADER 감성 분석을 수행하시오. (polarity_scores()를 사용해 감성 점수 계산)

In [ ]:
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer

# NLTK의 VADER 감성 분석 도구 초기화
nltk.download('vader_lexicon')
sid = SentimentIntensityAnalyzer()

In [5]:
# VADER를 사용하여 감성 분석 수행
scores = sid.polarity_scores(hamlet)

# 결과 출력
print(scores)

{'neg': 0.074, 'neu': 0.808, 'pos': 0.117, 'compound': 1.0}


neg 부정적 : 0.074, neu 중립적 : 0.808, pos 긍정적 : 0.117, coumpound 종합 감성 지수 : 1.0

-> 종합 감정 점수가 높을수록 긍정적 -> 긍정적

### (2) 문제 2 : 주어진 텍스트 데이터를 vader_polarity 함수를 사용하여 특정 임계값을 기준으로 긍정적 또는 부정적 감성으로 분류하시오.

In [6]:
# vader_polarity 함수 정의
def vader_polarity(text, threshold=0.1):
    scores = sid.polarity_scores(text)
    agg_score = scores["compound"]
    final_sentiment = 1 if agg_score >= threshold else 0
    return final_sentiment

# vader_polarity 함수를 이용하여 감성 분석 수행
hamlet_sentiment = vader_polarity(hamlet, threshold=0.1)

# 결과 출력
if hamlet_sentiment == 1:
    print("긍정적인 감성")
else:
    print("부정적인 감성")

긍정적인 감성


# 2. 조원들과 만든 문제와 답을 공유해보세요.